### PR TITLE
feat(add-rtl-layout-support): drawer, tabs, list items & about

### PR DESCRIPTION
**Pull request by Royosef**
#### What is it?
- [ ] Bug fix (user facing)
- [x] Feature (user facing)
- [ ] Code base improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

feat(add-rtl-layout-support): drawer, tabs, list items & about
* supportsRtl="true"
* replace Left/ Right properties with Start/ End
* tabs:
	* layoutDirection="ltr" for <TabsLayout>
	* reverse items for RTL in PagerAdapter
* properties for force right in RTL:
	* textAlignment="viewStart"
	* textDirection="anyRtl"

image

image

image

image

image

image

image

fixes https://github.com/TeamNewPipe/NewPipe/issues/3882

#### Testing apk

[NewPipe_AddRtlLayoutSupport-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/4533982/NewPipe_AddRtlLayoutSupport-debug.zip)

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/about/AboutActivity.java [hunk 1]
```diff
@@ -195,13 +209,21 @@ public int getCount() {
...
+            if (newPosition == 0) {
+                return getString(R.string.tab_about);
+            } else if (newPosition == 1) {
+                return getString(R.string.tab_licenses);
             }
+
             return null;
         }
+
+        private int getLayoutPosition(final int position) {
```
Comment (B0pol) [inline app/src/main/java/org/schabi/newpipe/about/AboutActivity.java:223]: As this method is duplicated thrice, should make it static util function, `int getLayoutPosition(boolean isRTL, int count, int position)`

File: app/src/main/java/org/schabi/newpipe/about/AboutActivity.java [hunk 2]
```diff
@@ -92,16 +93,24 @@ protected void onCreate(final Bundle savedInstanceState) {
...
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-        // Create the adapter that will return a fragment for each of the three
-        // primary sections of the activity.
-        mSectionsPagerAdapter = new SectionsPagerAdapter(getSupportFragmentManager());
 
         // Set up the ViewPager with the sections adapter.
         mViewPager = findViewById(R.id.container);
+
+        Configuration config = getResources().getConfiguration();
+        boolean isRTL = config.getLayoutDirection() == View.LAYOUT_DIRECTION_RTL;
```
Comment (B0pol) [inline app/src/main/java/org/schabi/newpipe/about/AboutActivity.java:101]: As this method is duplicated thrice, you should make it a static util method `boolean isRTL(Context context)` and then use `context.getResources()`.

**Comment by TobiGr** [comment]
> I'd merge LocalizeLayoutUtils (used to know if the device should be RTL and one method around that) and AndroidTvUtils to DeviceUtils. 

@B0pol Sounds good to me

**Comment by TobiGr** [comment]
@Royosef What's the state of this PR? Are you going to merge `LocalizeLayoutUtils` and `AndroidTvUtils` into `DeviceUtils`?

**Comment by wb9688** [comment]
I would move this to v0.20.0 as well

**Comment by heberjeur** [comment]
@B0pol @Stypox when this feature will be added?

**Comment by Stypox** [comment]
@heberjeur when @Royosef implements the requested changes and all bugs are fixed

**Comment by litetex** [comment]
No progress since a year, no GitHub actions build and a lot of merge conflicts.

Closing this issue for now. Feel free to reopen it when there is progress again.

**Pull request closed without merging**